app/src/main/java/io/intrepid/webservicesexercise/network/GithubClient.java

```diff
@@ -0,0 +1,53 @@
+package io.intrepid.webservicesexercise.network;
+
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import io.intrepid.webservicesexercise.model.GitHubUser;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Created by abassbayo-awoyemi on 1/11/17.
+ */
+
+public class GithubClient {
+
+    private static final long CONNECTION_TIMEOUT = 30;
+    private static final String BASE_URL = "https://api.github.com/users/";
+
+    private static GithubClient instance;
+    private GithubService service;
+
+    public static GithubClient getInstance(){
+        if(instance == null) {
```

Comment:
if the service instance is null, then go ahead and create it; as part of creating it, you'll instantiate the client